aten/src/ATen/ExpandUtils.cpp

```diff
@@ -21,7 +21,7 @@ std::vector<int64_t> infer_size(IntList a, IntList b) {
         ") must match the size of tensor b (", sizeB,
         ") at non-singleton dimension ", i);
 
-    expandedSizes[i] = std::max(sizeA, sizeB);
+    expandedSizes[i] = sizeA == 1 ? sizeB : sizeA;
```

Comment:
Bugfix?

Reply:
Yes (I think, the old behavior may have been there to preserve TH-style semantics, although I don't remember off the top of my head if that was intentional or not).  Anyway, I have the same fix in one of my n-dimensional 0-size branches and tests.